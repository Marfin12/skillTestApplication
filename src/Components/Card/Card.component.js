import * as React from 'react';
import { TouchableOpacity } from 'react-native';
import { useSelector } from 'react-redux';

import styles from './Card.styles';
import config from './Card.config';
import type { Props } from './Card.types';

const Card = (props: Props): React.Node => {
  const getSelect = useSelector(({ themeReducer }) => themeReducer);
  const { theme } = getSelect;

  return (
    <TouchableOpacity
      style={[styles.container(theme), props.style]}
      onPress={() => props.onPress()}
    >
      {props.children}
    </TouchableOpacity>
  );
};

Card.displayName = config.displayName;
Card.defaultProps = config.defaultProps;

export default Card;
