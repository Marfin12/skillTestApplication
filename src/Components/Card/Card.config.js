// @flow

import type { Props } from './Card.types';

import Constants from '../../Constants';

const defaultProps: Props = {
  children: null,
  style: {},
  onPress: Constants.UTILS.EMPTY_FUNCTION,
};

const displayName = 'Card';

export default {
  defaultProps,
  displayName,
};
