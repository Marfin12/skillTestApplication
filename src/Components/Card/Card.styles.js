import colors from "../../Colors";

const styles = {
  container: whiteTheme => {
    return {
      flex: 1,
      flexDirection: 'row',
      borderRadius: 5,
      backgroundColor: whiteTheme ? colors.blackPrimary : colors.whitePrimary,
      padding: 10,
      marginTop: 8,
      marginBottom: 8,
      elevation: 2,
    };
  },
};

export default styles;
