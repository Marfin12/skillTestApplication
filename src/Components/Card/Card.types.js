// @flow

import * as React from 'react';
import type {
  ViewStyleProp
} from 'react-native/Libraries/StyleSheet/StyleSheet';

export type Props = {|
  children: React.Node,
  style: ViewStyleProp,
  onPress: () => void
|};
