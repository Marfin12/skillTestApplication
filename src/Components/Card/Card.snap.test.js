import React from 'react';
import renderer from 'react-test-renderer';

import Card from './Card.component';
import config from './Card.config';

jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux');
  const lightMode = {
    theme: false,
  };
  const darkMode = {
    theme: true,
  };
  return {
    ...ActualReactRedux,
    useSelector: jest
      .fn()
      .mockImplementationOnce(() => {
        return lightMode;
      })
      .mockImplementationOnce(() => {
        return darkMode;
      }),
  };
});
jest.mock('../../Components/TextField', () => 'TextField');

afterEach(() => {
  jest.resetAllMocks();
});

describe('Card Component Default Theme', () => {
  const ComponentWrapper = renderer
    .create(<Card {...config.defaultProps} />)
    .toJSON();

  test('renders card component light mode', () => {
    expect(ComponentWrapper).toMatchSnapshot();
  });
});

describe('Card Component Dark Theme', () => {
  const ComponentWrapper = renderer
    .create(<Card {...config.defaultProps} />)
    .toJSON();

  test('renders card component dark mode', () => {
    expect(ComponentWrapper).toMatchSnapshot();
  });
});
