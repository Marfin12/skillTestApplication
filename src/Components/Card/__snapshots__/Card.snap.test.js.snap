// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component Dark Theme renders card component dark mode 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#222",
      "borderRadius": 5,
      "elevation": 2,
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 8,
      "marginTop": 8,
      "opacity": 1,
      "padding": 10,
    }
  }
/>
`;

exports[`Card Component Default Theme renders card component light mode 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFF",
      "borderRadius": 5,
      "elevation": 2,
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 8,
      "marginTop": 8,
      "opacity": 1,
      "padding": 10,
    }
  }
/>
`;
